src/test/java/org/reaktivity/nukleus/tcp/internal/streams/rfc793/ServerResetAndAbortIT.java

```diff
@@ -174,6 +174,48 @@ public void shouldShutdownInputWhenServerSendsReset() throws Exception
         }
     }
 
+    @Ignore("BEGIN vs RESET read order not yet guaranteed to match write order")
+    @Test
+    @Specification({
+        "${route}/server/controller",
+        "${server}/server.sent.reset.and.end/server"
+    })
+    @BMRule(name = "shutdownInput",
+    targetClass = "^java.nio.channels.SocketChannel",
+    targetMethod = "shutdownInput()",
+    helper = "org.reaktivity.nukleus.tcp.internal.SocketChannelHelper$CountDownHelper",
+    condition =
+      "callerMatches(\"org.reaktivity.nukleus.tcp.internal.stream.ReadStream..*\", true, true)",
+      action = "countDown()"
+    )
+    public void shouldShutdownInputWhenServerSendsResetAndEnd() throws Exception
+    {
+        CountDownLatch shutdownInputCalled = new CountDownLatch(1);
+        CountDownHelper.initialize(shutdownInputCalled);
+        k3po.start();
+        k3po.awaitBarrier("ROUTED_SERVER");
+
+        try (SocketChannel channel = SocketChannel.open())
+        {
+            channel.connect(new InetSocketAddress("127.0.0.1", 0x1f90));
 
+            channel.configureBlocking(false);
+
+            channel.write(ByteBuffer.wrap("some data".getBytes()));
```

Comment:
Similarly this test method should look really the same as shouldShutdownOutputAndInputWhenServerSendsAbortAndReset, and should be named accordingly (I think you should  be able to detect that output was shut down by reading -1 as in the other test, you should also eliminate the median writing of data since it adds nothing to the test)

Reply:
Ok, done.